test/StreamableBodyTest.cpp

```diff
@@ -82,3 +82,30 @@ TEST(StreamableBodyTest, TestGeneratorReturnsBufferedResults) {
   ASSERT_EQ(toStringView(result2->first), std::string("1Abc"));
   ASSERT_FALSE(result2->second);
 }
+
+TEST(StreamableBodyTest, TestHeadersAreSetCorrectlyForCompressingGenerator) {
+  namespace http = boost::beast::http;
+  using ad_utility::content_encoding::CompressionMethod;
+
+  auto generatorNone = generateNothing();
+  generatorNone.setCompressionMethod(CompressionMethod::NONE);
+  http::header<false, http::fields> headerNone;
+  streamable_body::writer{headerNone, generatorNone};
+
+  // empty string_view means no such header is present
+  ASSERT_EQ(headerNone[http::field::content_encoding], std::string_view{});
+
+  auto generatorDeflate = generateNothing();
+  generatorDeflate.setCompressionMethod(CompressionMethod::DEFLATE);
+  http::header<false, http::fields> headerDeflate;
+  streamable_body::writer{headerDeflate, generatorDeflate};
+
+  ASSERT_EQ(headerDeflate[http::field::content_encoding], "deflate");
+
+  auto generatorGzip = generateNothing();
+  generatorGzip.setCompressionMethod(CompressionMethod::GZIP);
+  http::header<false, http::fields> headerGzip;
+  streamable_body::writer{headerGzip, generatorGzip};
+
+  ASSERT_EQ(headerGzip[http::field::content_encoding], "gzip");

```

Comment:
You could factor out the common logic, by using the three pairs (NONE, ""),(DEFLATE, "deflate"), and (GZIP, "gzip")